safehttp/plugins/htmlinject/htmlinject.go

```diff
@@ -106,6 +121,85 @@ func Transform(src io.Reader, cfg ...Config) (string, error) {
 	return rw.out.String(), nil
 }
 
+// LoadTrustedTemplate processes the given TrustedTemplate with the specified default configurations and
+// adds it to the given template.
+// If the given template is nil a new one is created.
+func LoadTrustedTemplate(tpl *template.Template, csp, xsrf bool, src template.TrustedTemplate) (*template.Template, error) {
+	// Using bools to select configs prevents users to accidentally specify arbitrary and potentially unsafe rules.
+	var cfg []Config
+	noop := func() string {
+		panic("this function should never be called, templates should be cloned and injected with the noncing functions, not executed directly")
+	}
+	funcMap := map[string]interface{}{}
+	if csp {
+		cfg = append(cfg, CSPNoncesDefault)
+		funcMap[CSPNoncesDefaultFuncName] = noop
+	}
+	if xsrf {
+		cfg = append(cfg, XSRFTokensDefault)
+		funcMap[XSRFTokensDefaultFuncName] = noop
+	}
+	got, err := Transform(strings.NewReader(src.String()), cfg...)
+	if err != nil {
+		return nil, err
+	}
+	// We took a TrustedTemplate and transformed it with rules that we trust, so we know the output is still trusted.
+	tt := uncheckedconversions.TrustedTemplateFromStringKnownToSatisfyTypeContract(got)
+	if tpl == nil {
+		tpl = template.New("htmlinjected")
+	}
+	return tpl.Funcs(funcMap).ParseFromTrustedTemplate(tt)
+}
+
+// LoadGlob matches the behavior of safehtml.ParseGlob but runs a transformation on every loaded template.
+func LoadGlob(tpl *template.Template, csp, xsrf bool, pattern template.TrustedSource) (*template.Template, error) {
+	filenames, err := filepath.Glob(pattern.String())
+	if err != nil {
+		return nil, err
+	}
+	if len(filenames) == 0 {
+		return nil, fmt.Errorf("htmlinject: pattern matches no files: %#q", pattern.String())
```

Comment:
The `htmlinject:` part is known to the caller, so it should probably be omitted here.

Reply:
Other places too.